src/main/java/com/khomishchak/apigateway/validator/RouterValidator.java

```diff
@@ -0,0 +1,19 @@
+package com.khomishchak.apigateway.validator;
+
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+@Component
+public class RouterValidator {
+
+    public static final List<String> openApiEndpoints = List.of(
+            "/auth/register", "/auth/login", "/users", "/users/authenticate", "/exchangers/used-currencies"
+    );
+
+    public Predicate<ServerHttpRequest> isSecured =
+            request -> openApiEndpoints.stream()
+                    .noneMatch(uri -> request.getURI().getPath().contains(uri));
```

Comment:
The method name isSecured might be a bit misleading because it seems to check if the endpoint is not in the list of open API endpoints. I would recommend u to renaming it to something like isPublicEndpoint to better convey its intent BLYAT